File: HW#01/src/Unal_Baris.lua

--[[
****************************************************************************

CS315 HW01 - ASSOCIATIVE ARRAYS - LUA
Barış Tan Ünal
22003617
Section: 03

****************************************************************************
--]]

print("Hello, world!")

-- 1. Initialize
print "\n";
movieRatings = {};
movieRatings[ "Pulp Fiction" ] = 8;
movieRatings[ "The Green Mile" ] = 2;
movieRatings[ "Top Gun" ] = 7;
for movie, rating in pairs( movieRatings ) do
  print( movie, ": ", rating ); 
end

-- 2. Get the value for a given key
print "\n";
print( "- Rating of Top Gun: ", movieRatings[ "Top Gun" ] );

-- 3. Add a new element
print "\n";
movieRatings[ "Rear Window" ] = 10;
for movie, rating in pairs( movieRatings ) do
  print( movie, ": ", rating ); 
end

-- 4. Remove an element
print "\n";
movieRatings[ "Pulp Fiction" ] = nil;
for movie, rating in pairs( movieRatings ) do
  print( movie, ": ", rating ); 
end

-- 5. Modify the value of an existing element
print "\n";
movieRatings[ "The Green Mile" ] = 1;
for movie, rating in pairs( movieRatings ) do
  print( movie, ": ", rating ); 
end

-- 6. Search for the existence of a key
print "\n";
function keyExists( table, key )
  for movie, rating in pairs( table ) do
      if movie == key then return true; end
    end
  return false
end
print ( "Existence of Pulp Fiction: ", keyExists( movieRatings, "Pulp Fiction" ) );
print ( "Existence of Rear Window: ",keyExists( movieRatings, "Rear Window" ) );

-- 7. Search for the existence of a value
print "\n";
function valueExists( table, value )
  for movie, rating in pairs( table ) do
      if rating == value then return true; end
    end
  return false
end
print ( "Existence of a movie with 10/10 rating: ", valueExists( movieRatings, 10 ) );
print ( "Existence of a movie with 5/10 rating: ", valueExists( movieRatings, 5 ) );

-- 8. Loop through an associative array, apply a function, called foo, which simply prints the key-value pair
print "\n";
function foo( table ) 
    for movie, rating in pairs( table ) do
      print( movie, ": ", rating ); 
    end
end
foo( movieRatings );

File: HW#01/src/Unal_Baris.php
/*
****************************************************************************

CS315 HW01 - ASSOCIATIVE ARRAYS - PHP
Barış Tan Ünal
22003617
Section: 03

****************************************************************************
*/

<?php 
  
  echo "Hello, World!";

  // 1. Initialize
  $capitals = array( 
    "Brazil"=>"Brasilia", 
    "Russia"=>"Moscow", 
    "Thailand"=>"Bangkok", 
    "Cuba"=>"Havana", 
    "India"=>"New Delhi",
    "Malaysia"=>"Putrajaya"
  );
  print_r( "\nInitialized 'capitals':\n" );
  print_r( $capitals );
  
  // 2. Get the value for a given key
  print_r( "\nCapital of Cuba is: " ); 
  print_r( $capitals[ "Cuba" ] );
  
  // 3. Add a new element
  $capitals += array("Argentina" => "Buenos Aires");
  print_r( "\n\nAdded Argentina:\n" ); 
  print_r( $capitals );
  
  // 4. Remove an element
  unset( $capitals["Brazil"] );
  print_r( "\nRemoved Brazil:\n" ); 
  print_r( $capitals );
  
  // 5. Modify the value of an existing element
  $capitals[ 'Malaysia' ] = "Kuala Lumpur";
  print_r( "\nUpdated Malaysia:\n" ); 
  print_r( $capitals );
  
  // 6. Search for the existence of a key
  if( array_key_exists( "Russia", $capitals ) ) {
    print_r( "\nRussia exists." );
  }
  else {
    print_r( "\nRussia does not exist." );
  }

  if( array_key_exists( "USA", $capitals ) ) {
    print_r( "\USA exists." );
  }
  else {
    print_r( "\nUSA does not exist." );
  }
  
  // 7. Search for the existence of a value
  if( array_search( "Kuala Lumpur", $capitals ) ) {
    print_r( "\nKuala Lumpur exists." );
  }
  else {
    print_r( "\nKuala Lumpur does not exist." );
  }

  if( array_search( "Budapest", $capitals ) ) {
    print_r( "\nBudapest exists." );
  }
  else {
    print_r( "\nBudapest does not exist." );
  }

  
  // 8. Loop through an associative array, apply a function, called foo, which simply prints the key-value pair

  function foo( $array ) {
    foreach( $array as $key => $value ) { 
      echo $key . ": " . $value . "\n"; 
    } 
  }
  print_r( "\n\nCalling foo...\n" );
  foo( $capitals );

?>